src/repr/src/scalar.rs

```diff
@@ -1497,6 +1497,8 @@ impl<'a> ScalarType {
                 value_type: t,
                 custom_oid,
             } => custom_oid.is_some() || t.is_custom_type(),
+            // TODO(phemberger): also need to check every field within the record
```

Comment:
Hm, this whole `is_custom_type` thing seems pretty sketchy. Just reading up on https://github.com/MaterializeInc/materialize/blob/6a29b8604b058a95cae4761db7bafe5d9c5d9483/src/sql/src/func.rs#L533-L624 and I'm not entirely sure I understand what's going on.

Reply:
The idea is that we don't treat two custom types as interoperable, so you can't e.g. concatenate two custom lists together because we cannot determine the appropriate return type. lmk if you'd like to go through it in more depth, but that is essentially the shape of the hoop we're jumping through here.